### PR TITLE
Add join table.

### PR DESCRIPTION
**Pull request by firstpj**
In this project, I perform the following activities:

Create a table named vets with the following columns:
id: integer (set it as autoincremented PRIMARY KEY)
name: string
age: integer
date_of_graduation: date

There is a many-to-many relationship between the tables species and vets: a vet can specialize in multiple species, and a species can have multiple vets specialized in it. Create a "join table" called specializations to handle this relationship.

There is a many-to-many relationship between the tables animals and vets: an animal can visit multiple vets and one vet can be visited by multiple animals. Create a "join table" called visits to handle this relationship, it should also keep track of the date of the visit.

Insert the following data for vets:
Vet William Tatcher is 45 years old and graduated Apr 23rd, 2000.
Vet Maisy Smith is 26 years old and graduated Jan 17th, 2019.
Vet Stephanie Mendez is 64 years old and graduated May 4th, 1981.
Vet Jack Harkness is 38 years old and graduated Jun 8th, 2008.

Insert the following data for specializations:
Vet William Tatcher is specialized in Pokemon.
Vet Stephanie Mendez is specialized in Digimon and Pokemon.
Vet Jack Harkness is specialized in Digimon.

Insert the following data for visits:
Agumon visited William Tatcher on May 24th, 2020.
Agumon visited Stephanie Mendez on Jul 22th, 2020.
Gabumon visited Jack Harkness on Feb 2nd, 2021.
Pikachu visited Maisy Smith on Jan 5th, 2020.
Pikachu visited Maisy Smith on Mar 8th, 2020.
Pikachu visited Maisy Smith on May 14th, 2020.
Devimon visited Stephanie Mendez on May 4th, 2021.
Charmander visited Jack Harkness on Feb 24th, 2021.
Plantmon visited Maisy Smith on Dec 21st, 2019.
Plantmon visited William Tatcher on Aug 10th, 2020.
Plantmon visited Maisy Smith on Apr 7th, 2021.
Squirtle visited Stephanie Mendez on Sep 29th, 2019.
Angemon visited Jack Harkness on Oct 3rd, 2020.
Angemon visited Jack Harkness on Nov 4th, 2020.
Boarmon visited Maisy Smith on Jan 24th, 2019.
Boarmon visited Maisy Smith on May 15th, 2019.
Boarmon visited Maisy Smith on Feb 27th, 2020.
Boarmon visited Maisy Smith on Aug 3rd, 2020.
Blossom visited Stephanie Mendez on May 24th, 2020.
Blossom visited William Tatcher on Jan 11th, 2021.

Write queries to answer the following:
Who was the last animal seen by William Tatcher?
How many different animals did Stephanie Mendez see?
List all vets and their specialties, including vets with no specialties.
List all animals that visited Stephanie Mendez between April 1st and August 30th, 2020.
What animal has the most visits to vets?
Who was Maisy Smith's first visit?
Details for most recent visit: animal information, vet information, and date of visit.
How many visits were with a vet that did not specialize in that animal's species?
What specialty should Maisy Smith consider getting? Look for the species she gets the most

Screenshots
image
image
image
image
image
image
image
image
image
image

### CODE REVIEW

**Review by Shedrack-Sunday — CHANGES_REQUESTED** [review]
Hi @firstpj  

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

# Invalid Code Review Request 🔴 

You have submitted a project, that still has one of the sections that shows direction which is signified with a `>` sign. Kindly delete it and add your customized words

image

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: schema.sql [hunk 1]
```diff
@@ -1,14 +1,55 @@
...
+);
+CREATE TABLE species (
+    id serial PRIMARY KEY,
+    name varchar(255)
+);
 ALTER TABLE animals
-ADD species VARCHAR(100);
\ No newline at end of file
+ALTER COLUMN id ADD GENERATED ALWAYS AS IDENTITY;
+ALTER TABLE animals
```
Comment (skyv26) [inline schema.sql:26]: - [ ] Please double-check your work before asking for a review. I checked your queries and it seems like you have some issues while altering the table `animals`. Kindly look at the below error and fix it to have error-free queries.

image

File: queries.sql
Comment (skyv26) [inline queries.sql:1]: - [ ] As I mentioned above, it is very important to double-check the work. I tried your queries and got the result but the data that I got have some issues related to `species`. Either you have some issues with your queries or you missed adding the query in your file, which will help any person to create or `re-create` the entire database. I am getting some errors related to `visits` table or getting `unknown` species type.

**Review by skyv26 — CHANGES_REQUESTED** [review]
## Hi @firstpj, 

Good job so far 👍🏻 
There are some issues that you still need to work on to go to the next project but you are almost there!

## STATUS: REQUIRE CHANGES 🔴 



## Highlights
- Good **PR** title ✅ 
- Descriptive *PR* description ✅ 
- Implement the features ✅ 
- Added the `queries`, `data` and `schema` ✅

## Required Changes ♻️

- [ ] Kindly remove the `comments` from your `README` file, because those comments are meant for the developer as a guidance not for the normal user who will read your documentation and will get confuse. Please fix the things as I mentioned and `re-submit` your project.

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._


#### Cheers, and Happy coding!👏👏👏

## ⚠️  WARNING ⚠️ 
**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

## Have any doubt ❓ 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @skyv26 in your question so I can receive the notification. You can also follow me on the below platforms

<table>
<tr>
<td>

<a href="https://github.com/skyv26">
   @skyv26
</a>

</td>
<td>

<a href="https://www.linkedin.com/in/skyv2022/">
   linkedin-skyv2022
</a>

</td>
<td>

<a href="https://twitter.com/vrma_aakash">
   @vrma_aakash
</a>

</td>
<td>

<a href="https://www.instagram.com/skybrel/">
   @skybrel
</a>

</td>
<td>

<a href="https://www.frontendmentor.io/profile/skyv26">
   @skybrel
</a>

</td>
</tr>
</table>

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: schema.sql (hunk 1 above)
Comment (zeff96) [reply schema.sql:26]: - [ ] As mentioned by the previous reviewer, Kindly fix this error before submitting again

File: queries.sql
Comment (zeff96) [reply queries.sql:1]: - [ ] As mentioned by the previous reviewer, Kindly fix before submitting again

**Review by zeff96 — CHANGES_REQUESTED** [review]
### Status: CHANGES REQUIRED ♻️ ♻️ ♻️ 
Hi @firstpj , 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

almost

 ## Highlights
:heavy_check_mark: Correct GitFlow
:heavy_check_mark: Descriptive pull request

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by firstpj** [comment]
Thanks for the observations Sent from my Huawei Mobile-------- Original Message --------Subject: Re: [firstpj/vet-clinic] Add join table. (PR #4)From: Aakash Verma To: firstpj/vet-clinic CC: Collins Akorda ,Mention 
@skyv26 requested changes on this pull request.

Hi @firstpj,
Good job so far 👍🏻
There are some issues that you still need to work on to go to the next project but you are almost there!
STATUS: REQUIRE CHANGES 🔴

Highlights

Good PR title ✅
Descriptive PR description ✅
Implement the features ✅
Added the queries, data and schema ✅

Required Changes ♻️

 Kindly remove the comments from your README file, because those comments are meant for the developer as a guidance not for the normal user who will read your documentation and will get confuse. Please fix the things as I mentioned and re-submit your project.

Check the comments under the review.
Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better.
Cheers, and Happy coding!👏👏👏
⚠️  WARNING ⚠️
Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.
Have any doubt ❓
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @skyv26 in your question so I can receive the notification. You can also follow me on the below platforms




   




   




   




   




   





As described in the Code reviews limits policy you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using this form.

In schema.sql:
> +ALTER COLUMN id ADD GENERATED ALWAYS AS IDENTITY;
+ALTER TABLE animals


 Please double-check your work before asking for a review. I checked your queries and it seems like you have some issues while altering the table animals. Kindly look at the below error and fix it to have error-free queries.





On queries.sql:

 As I mentioned above, it is very important to double-check the work. I tried your queries and got the result but the data that I got have some issues related to species. Either you have some issues with your queries or you missed adding the query in your file, which will help any person to create or re-create the entire database. I am getting some errors related to visits table or getting unknown species type.


—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you were mentioned.Message ID: ***@***.***>
[
{
***@***.***": "http://schema.org",
***@***.***": "EmailMessage",
"potentialAction": {
***@***.***": "ViewAction",
"target": "https://github.com/firstpj/vet-clinic/pull/4#pullrequestreview-1651592011",
"url": "https://github.com/firstpj/vet-clinic/pull/4#pullrequestreview-1651592011",
"name": "View Pull Request"
},
"description": "View this Pull Request on GitHub",
"publisher": {
***@***.***": "Organization",
"name": "GitHub",
"url": "https://github.com"
}
}
]

**Review by omar-muhamad — APPROVED** [review]
# Approved ✔
### Hi @firstpj,

#### Great job, Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

### Congratulations! 🎉 

giphy (1)

## (Highlights) Good Points: :+1:
- Your work is well documented. ✔
- The correct workflow is used.  ✔
- All linters checks are ok. ✔

## (Changes Required) Aspects to improve: ♻️

- N/A

## [Optional] suggestions:

- N/A

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification**

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**